client/PerformanceMonitor.zig

```diff
@@ -0,0 +1,70 @@
+const std = @import("std");
+const renderite = @import("renderite");
+
+const log = std.log.scoped(.perf);
+
+const PerformanceMonitor = @This();
+
+state: renderite.Shared.PerformanceState,
+last_update: i128,
+last_frame: i128,
+counter: u32,
+
+pub fn init() PerformanceMonitor {
+    return .{
+        .state = .{
+            .fps = 0,
+            .immediateFPS = 0,
+            .renderTime = 0,
+            .externalUpdateTime = 0,
+            .frameBeginToSubmitTime = 0,
+            .frameProcessedToNextBeginTime = 0,
+            .integrationProcessingTime = 0,
+            .extraParticleProcessingTime = 0,
+            .processedAssetIntegratorTasks = 0,
+            .integrationHighPriorityTasks = 0,
+            .integrationTasks = 0,
+            .integrationRenderTasks = 0,
+            .integrationParticleTasks = 0,
+            .processingHandleWaits = 0,
+            .frameUpdateHandleTime = 0,
+            .renderedCameras = 0,
+            .renderedCameraPortals = 0,
+            .updatedTextures = 0,
+            .textureSliceUploads = 0,
+        },
+        .counter = 0,
+        .last_update = timestamp(),
+        .last_frame = timestamp(),
+    };
+}
+
+const time_base = @as(comptime_float, @floatFromInt(std.time.ns_per_s));
```

Comment:
I'm pretty sure in all the usages here the `comptime_int` should naturally coerce to the relevant float types

```suggestion
const time_base = std.time.ns_per_s;
```

Reply:
Nah, it causes problems:

```
client\PerformanceMonitor.zig:43:31: error: ambiguous coercion of division operands 'comptime_float' and 'comptime_int'; non-zero remainder '1'
const time_base_inverse = 1.0 / time_base;
                          ~~~~^~~~~~~~~~~
```

Best to just leave it.

Reply:
Should probably be
```
const time_base: comptime_float = ns_per_s;
``` 
then

Reply:
Oh, good call.